assets/js/modules/tagmanager/components/settings/SettingsForm.js

```diff
@@ -19,17 +19,42 @@
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import {
 	AccountSelect,
+	AMPContainerNameTextField,
 	AMPContainerSelect,
 	FormInstructions,
 	UseSnippetSwitch,
+	WebContainerNameTextField,
 	WebContainerSelect,
 } from '../common';
 import StoreErrorNotice from '../../../../components/StoreErrorNotice';
-import { STORE_NAME } from '../../datastore/constants';
+import { CONTAINER_CREATE, STORE_NAME } from '../../datastore/constants';
+const { useSelect } = Data;
 
 export default function SettingsForm() {
+	const containerID = useSelect( ( select ) => select( STORE_NAME ).getContainerID() );
+	const ampContainerID = useSelect( ( select ) => select( STORE_NAME ).getAMPContainerID() );
+
+	let containerNames = null;
+	if ( containerID === CONTAINER_CREATE || ampContainerID === CONTAINER_CREATE ) {
+		const webContainerName = containerID === CONTAINER_CREATE
+			? <WebContainerNameTextField />
+			: null;
+
+		const ampContainerName = ampContainerID === CONTAINER_CREATE
+			? <AMPContainerNameTextField />
+			: null;
+
+		containerNames = (
+			<div className="googlesitekit-setup-module__inputs">
+				{ webContainerName }
+				{ ampContainerName }
+			</div>
+		);
+	}
```

Comment:
The scoping of these variables and the assignments I found a bit tough to follow here. I don't notice a big readability difference with replacing this block with

```js
const showContainerName = containerID === CONTAINER_CREATE || ampContainerID === CONTAINER_CREATE;
```

… and replacing `{ containerNames }` below with:

```js
{ showContainerName && (
	<div className="googlesitekit-setup-module__inputs">
		{ containerID === CONTAINER_CREATE && <WebContainerNameTextField /> }
		{ ampContainerID === CONTAINER_CREATE && <AMPContainerNameTextField /> }
	</div>
) }
```

Just a thought, let me know what you think 🙂

Reply:
I like the idea of using a small component for it. Updated.